atomic_reactor/plugins/pre_reactor_config.py

```diff
@@ -0,0 +1,363 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from __future__ import unicode_literals
+
+from atomic_reactor.plugin import PreBuildPlugin
+
+from collections import namedtuple
+import logging
+import yaml
+
+
+# Key used to store the config object in the plugin workspace
+WORKSPACE_CONF_KEY = 'reactor_config'
+
+
+def get_config(workflow):
+    """
+    Obtain configuration object
+    Does not fail
+
+    :return: ReactorConfig instance
+    """
+    try:
+        workspace = workflow.plugin_workspace[ReactorConfigPlugin.key]
+        return workspace[WORKSPACE_CONF_KEY]
+    except KeyError:
+        # The plugin did not run or was not successful: use defaults
+        return ReactorConfig()
+
+
+class ConfigBuilder(object):
```

Comment:
JSON is great for transferring a data object from one program to another. It's picky in its syntax, though, and doesn't allow comments.

YAML is easier to write and allows comments.

This file is likely to be hand-edited from time to time. It would quite likely benefit from comments.

Other formats also allow comments, for example INI, but I wanted something in common use that provided structured data.